### DISCUSSION TITLE
Implementation of ICCP/TASE.2 protocol parser using Zeek

### DISCUSSION THREAD

**Discussion by biswajitutil** [cisagov/parsnip#34] [General]
Hi,
I am not able to find any Zeek implementation for ICCP/TASE.2 or IEC60870.6 although it is a very widely used and known protocol for control center communication in SCADA network. Can this community help in writing zeek spicy parser for the same? 

Thanks

**Comment by Crubumble** [comment]
I agree that this parser is important. That why it is current under development. The parser always has to be set up since the communication settings are not equal in facilities. The full installation of the parser is going to be updated this month [here](https://github.com/DINA-community/icsnpp-iccp).

If you want to help us, please run [ICCP-Setup](https://github.com/DINA-community/ICCP-Setup) and share it with us (see discussion [cisagov/Malcolm](https://github.com/cisagov/Malcolm/discussions/730)).

Also, zeek is aware of our project [zeek #4752](https://github.com/zeek/zeek/discussions/4752).

**Reply by biswajitutil** [reply]
Thanks for the reply. Will it be written in c or zeek spicy parser support will also come?

**Reply by Crubumble** [reply]
It's already written. However, it's not a single Zeek plugin, but a bundle. There is a separate plugin for each protocol layer, such as session log or MMS. Take a look at the repositories mentioned above, especially ICCP Setup, if you have a server for testing.
Your question about Spicy and C may be answered in Zeek discussion #4752. tldr: Some plugins are in Spicy, others are in C.
